### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by 0boro**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+services:
+  spaceapi:
+    image: spaceapi
+    deploy:
+      restart_policy:
+        condition: on-failure
+      labels:
+        - "traefik.enable=true"
+        - "traefik.http.services.spaceapi.loadbalancer.server.port=8000" 
+        - "traefik.http.routers.spaceapi.rule=Host(`spaceapi.flipdot.dev`)"
```
Comment (rbnis) [inline docker-compose.yml]: For prod the host should be `api.flipdot.org`.

File: deploytoswarm.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - name: Ensure docker_stack pip deps are installed
+      pip:
+        name: jsondiff
+      when: "'label_swarm_manager' in group_names"
+
+    - name: Ensure spaceapi repo is present
+      git:
+        repo: https://github.com/flipdot/spaceapi
+        dest: /root/spaceapi/
+        version: Add_dockerfile
```
Comment (rbnis) [inline deploytoswarm.yml]: This only works for this pull request.
With the next PR the default branch should be used.

**Review by rbnis — COMMENTED** [review]
See comments

File: deploytoswarm.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    - name: Ensure DNS is set up
+      inwx.collection.dns:
+        domain: "flipdot.dev"
+        type: "A"
+        record: "spaceapi"
+        value: "{{ hostvars[groups['label_swarm_manager'][0]]['ansible_default_ipv4']['address'] }}"
+        solo: true
+        ttl: "3600"
+        username: "test_flipdot_test"
```
Comment (cfstras) [inline deploytoswarm.yml]: should the username not also be an ENV?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+        - "traefik.http.routers.spaceapi.tls=true"
+        - "traefik.http.routers.spaceapi.tls.certresolver=myresolver"
```
Comment (cfstras) [inline docker-compose.yml]: what's `myresolver` and where is it defined?

**Review by cfstras — CHANGES_REQUESTED** [review]

File: deployment/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+version: '3.7'
+services:
+  spaceapi:
+    build:
+      context: ..
+    restart: always
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.services.spaceapi.loadbalancer.server.port=8000"
+      - "traefik.http.routers.spaceapi.rule=Host(`api.${base_domain}`)"
```
Comment (cfstras) [inline deployment/docker-compose.yml]: maybe "api" should also be a variable?

**Review by cfstras — APPROVED** [review]

**Pull request merged**